src/storage/mod.rs

```diff
@@ -195,8 +195,16 @@ impl Storage {
     }
 
     pub(crate) fn exists_in_archive(&self, archive_path: &str, path: &str) -> Result<bool> {
-        let index = self.get_index_for(archive_path)?;
-        Ok(index.find_file(path).is_ok())
+        match self.get_index_for(archive_path) {
+            Ok(index) => Ok(index.find_file(path).is_ok()),
+            Err(err) => {
+                if err.downcast_ref::<PathNotFoundError>().is_some() {
```

Comment:
Is there a way to check this without downcasting, maybe by changing the error type in `get_index_for`? I'm worried this is fragile.

Reply:
Do you mean something like `Result<Option<bool>>`? or a custom error-enum that lets us differentiate better between these errors? 

In general, because I don't know: what is the limitation of `downcast_ref`, why is it fragile?